src/golang/lib/job/spec.go

```diff
@@ -279,6 +292,31 @@ func NewParamSpec(
 	}
 }
 
+func NewSystemMetricSpec(
+	name string,
+	storageConfig *shared.StorageConfig,
+	metadataPath string,
+	metricName string,
+	inputMetadataPaths []string,
+	outputContentPaths []string,
+	outputMetadataPaths []string,
+) Spec {
+	return &SystemMetricSpec{
+		basePythonSpec: basePythonSpec{
+			baseSpec: baseSpec{
+				Type: SystemMetricJobType,
+				Name: name,
+			},
+			StorageConfig: *storageConfig,
+			MetadataPath:  metadataPath,
+		},
+		InputMetadataPaths:  inputMetadataPaths,
+		OutputContentPaths:  outputContentPaths,
+		OutputMetadataPaths: outputMetadataPaths,
```

Comment:
wait it still doesn't make sense to me how there can be multiple output paths. There's only one metric per spec right? So it should be a single `OutputContentPath` and `OutputMetadataPath`.

Reply:
fixed variable name and struct decleration from string[] to string to enforce it